presto-main/src/test/java/com/facebook/presto/operator/TestHashJoinOperator.java

```diff
@@ -126,25 +130,34 @@ public void tearDown()
         scheduledExecutor.shutdownNow();
     }
 
-    @DataProvider(name = "hashEnabledValues")
-    public static Object[][] hashEnabledValuesProvider()
+    @DataProvider(name = "hashJoinTestValues")
+    public static Object[][] hashJoinTestValuesProvider()
     {
+        // parallelBuild, probeHashEnabled, buildHashEnabled, and isDictionaryProcessingJoinEnabled respectively
         return new Object[][] {
-                {true, true, true},
-                {true, true, false},
-                {true, false, true},
-                {true, false, false},
-                {false, true, true},
-                {false, true, false},
-                {false, false, true},
-                {false, false, false}};
+                {true, true, true, true},
+                {true, true, true, false},
+                {true, true, false, true},
+                {true, true, false, false},
+                {true, false, true, true},
+                {true, false, true, false},
+                {true, false, false, true},
+                {true, false, false, false},
+                {false, true, true, true},
+                {false, true, true, false},
+                {false, true, false, true},
+                {false, true, false, false},
+                {false, false, true, true},
+                {false, false, true, false},
+                {false, false, false, true},
+                {false, false, false, false}};
```

Comment:
Did we reach the state when following would be more readable:
```
ImmutableList.Builder<Object[]> values = ImmutableList.builder();
List<Boolean> trueFalse = ImmutableList.of(true,false);
for (parallelBuild : trueFalse) {
  for (probeHashEnabled : trueFalse) {
    for (buildHashEnabled : trueFalse) {
      for (isDictionaryProcessingJoinEnabled : trueFalse) {
        values.add(new Object[] { parallelBuild, probeHashEnabled, buildHashEnabled, isDictionaryProcessingJoinEnabled });
      }
    }
  }
}
return values.build().stream().toArray(Object[][]::new);
```
?

(otherwise someone reading this might doubts whether all combinations are covered)

Reply:
I now see https://github.com/prestodb/presto/pull/9423/commits/4077091c1e3de17a30824def8249f3c1a62159d7 which basically retracts this change.. disregard this comment